crates/datafusion_ext/src/planner/relation/mod.rs

```diff
@@ -261,10 +268,45 @@ fn infer_func_for_file(path: &str) -> Result<OwnedTableReference> {
             schema: "public".into(),
             table: "read_bson".into(),
         },
+        "gz" => {
+            //handling compressed files with .gz extension
+            infer_func_from_compressed_file(filename)?
+        }
         ext => {
             return Err(DataFusionError::Plan(format!(
                 "unable to infer how to handle file extension: {ext}"
             )))
         }
     })
 }
+
+fn infer_func_from_compressed_file(filename: &str) -> Result<OwnedTableReference> {
```

Comment:
> I think we want these conditions to be "ends with" rather than "contains"?
Yes, "ends with" is better way to ascertain the file's format. I will modify this.

>I'm pretty sure we also support bzip (.bz) and zstd (`.xz) compression methods.
I don't have adequate knowledge about different types we should support, But can surely add as required.

Thanks.